napari/_qt/dialogs/shimmed_plugin_dialog.py

```diff
@@ -0,0 +1,83 @@
+from qtpy.QtWidgets import (
+    QCheckBox,
+    QDialog,
+    QHBoxLayout,
+    QLabel,
+    QPushButton,
+    QVBoxLayout,
+    QWidget,
+)
+
+from napari.settings import get_settings
+from napari.settings._constants import PluginShimWarningLevel
+from napari.utils.translations import trans
+
+
+class ShimmedPluginDialog(QDialog):
+    def __init__(self, parent: QWidget, plugins: set[str]) -> None:
+        super().__init__(parent)
+        self.setWindowTitle('Installed Plugin Warning')
+        okay_btn = QPushButton(trans._('Okay'))
+        icon_label = QWidget()
+        icon_label.setObjectName('warning_icon_element')
+        self.plugin_info_text = """
+The following plugins use napari's old plugin engine. These plugins have been automatically converted to the new plugin engine, npe2:
+        """
+        self.plugins = plugins
+        self.plugin_text = '\n'.join(self.plugins)
```

Comment:
```suggestion
        self.plugin_text = '\n'.join(sorted(self.plugins))
```

For consistent order